### PR TITLE
Explore/Loki: Escape \ in labels for show context queries

### LINKED ISSUE #23337: Show context doesn't escape labels values when firing request.
**What happened**:

When querying a stream that contains windows style path such as `{filename="E:\\Logs\\try\\test\\test.log"} |= "foo"` This query works perfectly as the \ is escaped.

However when using the show context feature it will send a request using this : `{filename="E:\Logs\try\test\test.log"}` and this time \ are not escaped and will result into a 404

**What you expected to happen**:

Show context queries should escape \.

**How to reproduce it (as minimally and precisely as possible)**:

It's not easy to reproduce since you need windows style path.

**Environment**:
- Grafana version: Grafana v7.0.0-47498119pre (a1157b1744)

- Data source type & version: Loki

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

When we are creating log context query, we are creating it from labels-values that we  received for the chosen log row. Therefore user has no way to escape special characters with backslash as required by [LogQL](https://github.com/grafana/loki/blob/master/docs/logql.md). So for the context query, we need to escape it for the user within Grafana. 

Currently we are escaping `\` character, as when I tested  other characters,  queries worked without escaping - e.g. `[ $ ? * ( )` . 

image

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/grafana/issues/23337

**Special info for reviewer**:
I have temporarily added `script.go` file that generated logs with `\`. So to test this PR, run loki `make devenv sources=loki` and run `go run script.go`. You will then have access to `filenameA="E:\Logs\try\test\test.log` label that can be used for testing. For query field, you still need to escape `\` by yourself, so the query will be `{filenameA="E:\\Logs\\try\\test\\test.log}`.

The `script.go` file will be removed after this PR is approved and before merging to master.

### CODE REVIEW

File: script.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package main
```
Comment (aknuds1) [inline script.go:1]: Is this file included by accident?

Comment (ivanahuckova) [reply script.go:1]: As mentioned in the description ```Special info for reviewer:
I have temporarily added script.go file that generated logs with \. So to test this PR, run loki make devenv sources=loki and run go run script.go. You will then have access to filenameA="E:\Logs\try\test\test.log label that can be used for testing. For query field, you still need to escape \ by yourself, so the query will be {filenameA="E:\\Logs\\try\\test\\test.log}.

The script.go file will be removed after this PR is approved and before merging to master.```

File: public/app/plugins/datasource/loki/datasource.ts [hunk 2]
```diff
@@ -409,7 +409,7 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
 
   prepareLogRowContextQueryTarget = (row: LogRowModel, limit: number, direction: 'BACKWARD' | 'FORWARD') => {
     const query = Object.keys(row.labels)
-      .map(label => `${label}="${row.labels[label]}"`)
+      .map(label => `${label}="${row.labels[label].replace(/\\/g, '\\\\')}"`) // escape backslash in label as users can't escape it by themselves
```
Comment (aknuds1) [inline public/app/plugins/datasource/loki/datasource.ts:412]:
```suggestion
      .map(label => `${label}="${row.labels[label].replace(/\\/g, '\\\\')}"`) // escape backslashes in label as users can't escape them by themselves
```

**Review by aknuds1 — COMMENTED** [review]
I think the comment should refer to backslashes in plural.

File: public/app/plugins/datasource/loki/datasource.ts (hunk 2 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/datasource.ts:412]: Yes, makes sense! I agree.

File: script.go (hunk 1 above)
Comment (aknuds1) [reply script.go:1]: Yeah sorry, only saw it later in the PR description. Never mind.

**Review by aknuds1 — APPROVED** [review]

File: script.go (hunk 1 above)
Comment (ivanahuckova) [reply script.go:1]: No worries! I just wanted to make testing of this PR easier for the one who will be testing it. But I'll definitely remove the file before merging. 🙂

Comment (ivanahuckova) [inline script.go:1]: This file is only for PR testing purposes. Will be removed before merging.

**Review by aocenas — APPROVED** [review]

**Pull request merged**